src/Symfony/Component/VarDumper/Caster/ReflectionCaster.php

```diff
@@ -76,7 +76,18 @@ public static function castClosure(\Closure $c, array $a, Stub $stub, $isNested)
 
     public static function castGenerator(\Generator $c, array $a, Stub $stub, $isNested)
     {
-        return class_exists('ReflectionGenerator', false) ? self::castReflectionGenerator(new \ReflectionGenerator($c), $a, $stub, $isNested) : $a;
+        if (!class_exists('ReflectionGenerator', false)) {
+            return $a;
```

Comment:
I can not because it alters the generator. That's why I used the try catch method and not `->valid()`.